controllers/infinispan_controller.go

```diff
@@ -1363,6 +1445,7 @@ func (r *infinispanRequest) reconcileContainerConf(statefulSet *appsv1.StatefulS
 	// Validate ConfigMap changes (by the hash of the infinispan.yaml key value)
 	updateNeeded = updateStatefulSetEnv(statefulSet, "CONFIG_HASH", hash.HashString(configMap.Data[consts.ServerConfigFilename])) || updateNeeded
 	updateNeeded = updateStatefulSetEnv(statefulSet, "ADMIN_IDENTITIES_HASH", hash.HashByte(adminSecret.Data[consts.ServerIdentitiesFilename])) || updateNeeded
+	updateNeeded = updateStatefulSetEnv(statefulSet, "OVERLAY_CONFIG_HASH", hash.HashByte([]byte(overlayConfigMap.Data[overlayConfigMapKey]))) || updateNeeded
```

Comment:
```suggestion
	updateNeeded = updateStatefulSetEnv(statefulSet, "OVERLAY_CONFIG_HASH", hash.HashString(overlayConfigMap.Data[overlayConfigMapKey])) || updateNeeded
```

Reply:
But would be nice to implement #1077 and don't litter env variables in container with unnecessary info

Reply:
Agree, but overlay config map annotation would be introduced before release to avoid migration validation. Yes, others env -> annotation conversion need to be tested during migration